File: src/main/java/io/github/takenoha/towerdefense/config/CombatSettings.java
package io.github.takenoha.towerdefense.config;

/**
 * Spatial, participation, and phase timing limits for one defense encounter.
 * Distances are measured horizontally in blocks and times are measured in seconds.
 */
public record CombatSettings(
        double radius,
        double spawnInner,
        double spawnOuter,
        double minimumCoreDistance,
        double coreGap,
        int maxParticipants,
        int countdownSeconds,
        int preparationSeconds,
        int intermissionSeconds,
        int absenceGraceSeconds) {
}

File: src/main/kotlin/io/github/takenoha/towerdefense/config/CombatSettings.kt
package io.github.takenoha.towerdefense.config

/** Spatial, participation, and phase timing limits for one defense encounter. */
@JvmRecord
data class CombatSettings(
    val radius: Double,
    val spawnInner: Double,
    val spawnOuter: Double,
    val minimumCoreDistance: Double,
    val coreGap: Double,
    val maxParticipants: Int,
    val countdownSeconds: Int,
    val preparationSeconds: Int,
    val intermissionSeconds: Int,
    val absenceGraceSeconds: Int,
)
